lib/Connection.php

```diff
@@ -580,37 +592,26 @@ private function doConnect($host, $port) {
 		} else {
 			throw new ServerNotAvailableException('Could not set required LDAP Protocol version.');
 		}
+		if (!$this->getLDAP()->isResource($this->ldapConnectionRes)) {
+			$this->ldapConnectionRes = null; // to indicate it really is not set, connect() might have set it to false
+			throw new ServerNotAvailableException("Connect to $host:$port failed");
+		}
 		return true;
 	}
 
 	/**
 	 * Binds to LDAP
+	 * @throws \OC\ServerNotAvailableException
 	 */
 	public function bind() {
-		static $getConnectionResourceAttempt = false;
 		if(!$this->configuration->ldapConfigurationActive) {
 			return false;
 		}
-		if($getConnectionResourceAttempt) {
-			$getConnectionResourceAttempt = false;
-			return false;
-		}
-		$getConnectionResourceAttempt = true;
+
 		$cr = $this->getConnectionResource();
-		$getConnectionResourceAttempt = false;
 		if(!$this->getLDAP()->isResource($cr)) {
 			return false;
 		}
-		$ldapLogin = @$this->getLDAP()->bind($cr,
```

Comment:
Bind function not doing `bind()`? This needs a comment: binding is done via `getConnectionResource()` (if I'm not mistaken)